packages/@aws-cdk/core/test/private/tree-metadata.test.ts

```diff
@@ -106,15 +106,73 @@ nodeunitShim({
                     },
                   },
                 },
-              },
+              }),
             },
-          },
+          }),
         },
-      },
+      }),
     });
     test.done();
   },
 
+  'tree metadata has construct class & version in there'(test: Test) {
+    // The runtime metadata this test relies on is only available if the most
+    // recent compile has happened using 'jsii', as the jsii compiler injects
+    // this metadata.
+    //
+    // If the most recent compile was using 'tsc', the metadata will not have
+    // been injected, and the test will fail.
+    //
+    // People may choose to run `tsc` directly (instead of `yarn build` for
+    // example) to escape the additional TSC compilation time that is necessary
+    // to run 'eslint', or the additional time that 'jsii' needs to analyze the
+    // type system), this test is allowed to fail if we're not running on CI.
+    //
+    // If the compile of this library has been done using `tsc`, the runtime
+    // information will always find `constructs.Construct` as the construct
+    // identifier, since the copy of that library that is loaded here will
+    // always have had a release build done using 'jsii'.
```

Comment:
```suggestion
    // If the compile of this library has been done using `tsc`, the runtime
    // information will always find `constructs.Construct` as the construct
    // identifier, since `constructs` will have had a release build done using `jsii`.
```